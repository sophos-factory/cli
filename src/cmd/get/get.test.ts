/* eslint-disable @typescript-eslint/no-non-null-assertion */
import { execute } from '../../../tests/helpers/execute';
import knownIds from '../../../tests/helpers/knownIds';

describe('factoryctl get', () => {
  jest.setTimeout(60 * 1000);

  test('throws on missing subcommand', async () => {
    await expect(
      execute(['get'], { token: process.env.FACTORY_STATIC_AUTH_TOKEN! })
    ).rejects.toMatchSnapshot();
  });

  describe('organization', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['get', 'organization', '--format', 'json'], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();
    });
  });

  describe('project', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['get', 'project'], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();
    });
  });

  describe('credential', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['get', 'credential', '--format', 'json'], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();

      // Missing positional
      await expect(
        execute(['get', 'credential', '--project-id', knownIds.project], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();
    });
  });

  describe('job', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['get', 'job', '--format', 'json'], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();

      // Missing positional
      await expect(
        execute(
          ['get', 'job', '--project-id', knownIds.project, '--format', 'json'],
          {
            token: process.env.FACTORY_STATIC_AUTH_TOKEN!
          }
        )
      ).rejects.toMatchSnapshot();
    });
  });

  describe('pipeline-revision', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['get', 'pipeline-revision', '--format', 'json'], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();

      // Missing positional
      await expect(
        execute(
          [
            'get',
            'pipeline-revision',
            '--project-id',
            knownIds.project,
            '--pipeline-id',
            knownIds.pipeline,
            '--format',
            'json'
          ],
          { token: process.env.FACTORY_STATIC_AUTH_TOKEN! }
        )
      ).rejects.toMatchSnapshot();
    });
  });

  describe('pipeline', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['get', 'pipeline', '--format', 'json'], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();

      await expect(
        execute(
          [
            'get',
            'pipeline',
            '--project-id',
            knownIds.project,
            '--format',
            'json'
          ],
          {
            token: process.env.FACTORY_STATIC_AUTH_TOKEN!
          }
        )
      ).rejects.toMatchSnapshot();
    });
  });

  describe('runner', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['get', 'runner', '--format', 'json'], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();

      // Missing positional
      await expect(
        execute(['get', 'runner', '--organization-id', knownIds.organization], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();
    });
  });

  describe('runner-pool', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['get', 'runner-pool', '--format', 'json'], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();

      // Missing positional
      await expect(
        execute(
          ['get', 'runner-pool', '--organization-id', knownIds.organization],
          {
            token: process.env.FACTORY_STATIC_AUTH_TOKEN!
          }
        )
      ).rejects.toMatchSnapshot();
    });
  });

  describe('run', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['get', 'run', '--format', 'json'], {
          token: process.env.FACTORY_STATIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();

      // Missing positional
      await expect(
        execute(
          ['get', 'run', '--project-id', knownIds.project, '--format', 'json'],
          {
            token: process.env.FACTORY_STATIC_AUTH_TOKEN!
          }
        )
      ).rejects.toMatchSnapshot();
    });
  });
});
