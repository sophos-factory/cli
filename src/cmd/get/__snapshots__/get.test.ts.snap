// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`factoryctl get credential throws on missing arguments 1`] = `
"Usage: factoryctl get credential <credential-id> [options]

Positionals:
  credential-id  ID of the credential to fetch. Unlike IDs for other resources (which are mechanically generated), credential ID is user-provided reference ID  [string] [default: read from stdin]

Options:
      --version       Show version number  [boolean]
      --address       Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address  Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token    Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter        Filter output using JSONPath  [string]
      --format        Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose       Print detailed output  [boolean]
      --help          Show help  [boolean]
      --project-id    ID of the project containing the credential  [string] [required]

Missing required argument: project-id
"
`;

exports[`factoryctl get credential throws on missing arguments 2`] = `
"Usage: factoryctl get credential <credential-id> [options]

Positionals:
  credential-id  ID of the credential to fetch. Unlike IDs for other resources (which are mechanically generated), credential ID is user-provided reference ID  [string] [default: read from stdin]

Options:
      --version       Show version number  [boolean]
      --address       Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address  Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token    Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter        Filter output using JSONPath  [string]
      --format        Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose       Print detailed output  [boolean]
      --help          Show help  [boolean]
      --project-id    ID of the project containing the credential  [string] [required]

Credential ID must be provided
"
`;

exports[`factoryctl get job throws on missing arguments 1`] = `
"Usage: factoryctl get job <job-id> [options]

Positionals:
  job-id  ID of the job to fetch  [string] [default: read from stdin]

Options:
      --version       Show version number  [boolean]
      --address       Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address  Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token    Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter        Filter output using JSONPath  [string]
      --format        Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose       Print detailed output  [boolean]
      --help          Show help  [boolean]
      --project-id    ID of the project containing the job  [string] [required]

Missing required argument: project-id
"
`;

exports[`factoryctl get job throws on missing arguments 2`] = `
"Usage: factoryctl get job <job-id> [options]

Positionals:
  job-id  ID of the job to fetch  [string] [default: read from stdin]

Options:
      --version       Show version number  [boolean]
      --address       Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address  Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token    Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter        Filter output using JSONPath  [string]
      --format        Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose       Print detailed output  [boolean]
      --help          Show help  [boolean]
      --project-id    ID of the project containing the job  [string] [required]

Job ID must be provided
"
`;

exports[`factoryctl get organization throws on missing arguments 1`] = `
"Usage: factoryctl get organization <organization-id> [options]

Positionals:
  organization-id  ID of the organization to fetch  [string] [default: read from stdin]

Options:
      --version       Show version number  [boolean]
      --address       Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address  Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token    Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter        Filter output using JSONPath  [string]
      --format        Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose       Print detailed output  [boolean]
      --help          Show help  [boolean]

Organization ID must be provided
"
`;

exports[`factoryctl get pipeline throws on missing arguments 1`] = `
"Usage: factoryctl get pipeline <pipeline-id> [options]

Positionals:
  pipeline-id  ID of the pipeline to fetch  [string] [default: read from stdin]

Options:
      --version       Show version number  [boolean]
      --address       Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address  Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token    Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter        Filter output using JSONPath  [string]
      --format        Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose       Print detailed output  [boolean]
      --help          Show help  [boolean]
      --project-id    ID of the project to fetch  [string] [required]

Missing required argument: project-id
"
`;

exports[`factoryctl get pipeline throws on missing arguments 2`] = `
"Usage: factoryctl get pipeline <pipeline-id> [options]

Positionals:
  pipeline-id  ID of the pipeline to fetch  [string] [default: read from stdin]

Options:
      --version       Show version number  [boolean]
      --address       Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address  Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token    Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter        Filter output using JSONPath  [string]
      --format        Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose       Print detailed output  [boolean]
      --help          Show help  [boolean]
      --project-id    ID of the project to fetch  [string] [required]

Pipeline ID must be provided
"
`;

exports[`factoryctl get pipeline-revision throws on missing arguments 1`] = `
"Usage: factoryctl get pipeline-revision <revision> [options]

Positionals:
  revision-id  ID of the pipeline revision to fetch  [string] [default: read from stdin]

Options:
      --version       Show version number  [boolean]
      --address       Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address  Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token    Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter        Filter output using JSONPath  [string]
      --format        Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose       Print detailed output  [boolean]
      --help          Show help  [boolean]
      --project-id    ID of the project containing the pipeline revision  [string] [required]
      --pipeline-id   ID of the pipeline for the pipeline revision  [string] [required]

Missing required arguments: project-id, pipeline-id
"
`;

exports[`factoryctl get pipeline-revision throws on missing arguments 2`] = `
"Usage: factoryctl get pipeline-revision <revision> [options]

Positionals:
  revision-id  ID of the pipeline revision to fetch  [string] [default: read from stdin]

Options:
      --version       Show version number  [boolean]
      --address       Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address  Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token    Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter        Filter output using JSONPath  [string]
      --format        Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose       Print detailed output  [boolean]
      --help          Show help  [boolean]
      --project-id    ID of the project containing the pipeline revision  [string] [required]
      --pipeline-id   ID of the pipeline for the pipeline revision  [string] [required]

Revision must be provided
"
`;

exports[`factoryctl get project throws on missing arguments 1`] = `
"Usage: factoryctl get project <project-id> [options]

Positionals:
  project-id  ID of the project to fetch  [string] [default: read from stdin]

Options:
      --version       Show version number  [boolean]
      --address       Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address  Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token    Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter        Filter output using JSONPath  [string]
      --format        Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose       Print detailed output  [boolean]
      --help          Show help  [boolean]

Project ID must be provided
"
`;

exports[`factoryctl get run throws on missing arguments 1`] = `
"Usage: factoryctl get run <run-id> [options]

Positionals:
  run-id  ID of the run to fetch  [string] [default: read from stdin]

Options:
      --version       Show version number  [boolean]
      --address       Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address  Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token    Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter        Filter output using JSONPath  [string]
      --format        Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose       Print detailed output  [boolean]
      --help          Show help  [boolean]
      --project-id    ID of the project containing the run  [string] [required]
      --wait          Wait until run execution is finished if it is in progress  [boolean]

Missing required argument: project-id
"
`;

exports[`factoryctl get run throws on missing arguments 2`] = `
"Usage: factoryctl get run <run-id> [options]

Positionals:
  run-id  ID of the run to fetch  [string] [default: read from stdin]

Options:
      --version       Show version number  [boolean]
      --address       Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address  Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token    Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter        Filter output using JSONPath  [string]
      --format        Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose       Print detailed output  [boolean]
      --help          Show help  [boolean]
      --project-id    ID of the project containing the run  [string] [required]
      --wait          Wait until run execution is finished if it is in progress  [boolean]

Run ID must be provided
"
`;

exports[`factoryctl get runner throws on missing arguments 1`] = `
"Usage: factoryctl get runner <runner-id> [options]

Positionals:
  runner-id  ID of the runner to fetch  [string] [default: read from stdin]

Options:
      --version          Show version number  [boolean]
      --address          Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address     Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token       Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter           Filter output using JSONPath  [string]
      --format           Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose          Print detailed output  [boolean]
      --help             Show help  [boolean]
      --organization-id  ID of the organization containing the runner  [string]
      --project-id       ID of the project containing the runner  [string]

Runner ID must be provided
"
`;

exports[`factoryctl get runner throws on missing arguments 2`] = `
"Usage: factoryctl get runner <runner-id> [options]

Positionals:
  runner-id  ID of the runner to fetch  [string] [default: read from stdin]

Options:
      --version          Show version number  [boolean]
      --address          Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address     Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token       Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter           Filter output using JSONPath  [string]
      --format           Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose          Print detailed output  [boolean]
      --help             Show help  [boolean]
      --organization-id  ID of the organization containing the runner  [string]
      --project-id       ID of the project containing the runner  [string]

Runner ID must be provided
"
`;

exports[`factoryctl get runner-pool throws on missing arguments 1`] = `
"Usage: factoryctl get runner-pool <runner-pool-id> [options]

Positionals:
  runner-pool-id  ID of the runner pool to fetch  [string] [default: read from stdin]

Options:
      --version          Show version number  [boolean]
      --address          Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address     Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token       Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter           Filter output using JSONPath  [string]
      --format           Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose          Print detailed output  [boolean]
      --help             Show help  [boolean]
      --organization-id  ID of the organization containing the runner pool  [string] [required]

Missing required argument: organization-id
"
`;

exports[`factoryctl get runner-pool throws on missing arguments 2`] = `
"Usage: factoryctl get runner-pool <runner-pool-id> [options]

Positionals:
  runner-pool-id  ID of the runner pool to fetch  [string] [default: read from stdin]

Options:
      --version          Show version number  [boolean]
      --address          Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address     Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token       Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter           Filter output using JSONPath  [string]
      --format           Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose          Print detailed output  [boolean]
      --help             Show help  [boolean]
      --organization-id  ID of the organization containing the runner pool  [string] [required]

Runner Pool ID must be provided
"
`;

exports[`factoryctl get throws on missing subcommand 1`] = `
"Usage: factoryctl get <command> [options]

Commands:
  factoryctl get credential [credential-id]       Get credential details
  factoryctl get job [job-id]                     Get job details
  factoryctl get organization [organization-id]   Get organization details
  factoryctl get pipeline [pipeline-id]           Get pipeline details
  factoryctl get pipeline-revision [revision-id]  Get pipeline-revision details
  factoryctl get project [project-id]             Get project details
  factoryctl get run [run-id]                     Get run details
  factoryctl get runner [runner-id]               Get runner details
  factoryctl get runner-pool [runner-pool-id]     Get runner pool details

Options:
      --version       Show version number  [boolean]
      --address       Address of the Factory API server. Required for all commands except "organization(s)  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address  Address of the Factory Auth API server. Required for "organization(s)" commands  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token    Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter        Filter output using JSONPath  [string]
      --format        Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose       Print detailed output  [boolean]
      --help          Show help  [boolean]

Command must be specified
"
`;
